airgun/entities/subscription.py

```diff
@@ -0,0 +1,164 @@
+from navmazing import NavigateToSibling
+from wait_for import wait_for
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.views.subscription import (
+    AddSubscriptionView,
+    DeleteManifestConfirmationView,
+    ManageManifestView,
+    SubscriptionDetailsView,
+    SubscriptionListView,
+)
+
+
+class SubscriptionEntity(BaseEntity):
+    def _wait_for_process_to_finish(self, name, has_manifest=False):
+        view = self.navigate_to(self, 'All')
+        wait_for(
+                lambda: view.flash.assert_message(
+                        "Task {} completed".format(name), partial=True),
+                handle_exception=True, timeout=60*10,
+                logger=view.flash.logger
+        )
+        wait_for(
+                lambda: not view.progressbar.is_displayed,
+                handle_exception=True, timeout=60*10,
+                logger=view.progressbar.logger
+        )
+        wait_for(
+                lambda: self.has_manifest == has_manifest,
+                handle_exception=True, timeout=10,
+                logger=view.logger
+        )
+        view.flash.dismiss()
+
+    @property
+    def has_manifest(self):
```

Comment:
why do we need that property? Looks like redundant navigation as we can have `lambda: not view.add_button.disabled == has_manifest`

Reply:
Addressed in comment below: https://github.com/SatelliteQE/airgun/pull/163#issuecomment-427853002